Frameworks/CoreGraphics/D2DWrapper.mm

```diff
@@ -19,18 +19,32 @@
 
 using namespace Microsoft::WRL;
 
+namespace {
+template <typename T>
+struct HRComPtr {
+    HRESULT hr;
+    ComPtr<T> ptr;
+};
+}
+
 HRESULT _CGGetD2DFactory(ID2D1Factory** factory) {
-    static ComPtr<ID2D1Factory> sFactory;
```

Comment:
(there may have been an innate unsafeness here: I don't know if the static init order here is specified or if it's guaranteed that this `ComPtr` will construct before the static HRESULT is initialized.